demo/lib/vm.bash

```diff
@@ -174,6 +174,49 @@ vm-check-running-binary() {
     return 0
 }
 
+vm-wait-process-startup() {
+    # Usage: vm-wait-process-startup [--timeout TIMEOUT] [--pidfile PIDFILE] PROCESS
+    # Wait for a PROCESS (string) to appear in process list.
+    # If pidfile parameter is given, we also check that the process has the file open.
+    # The default TIMEOUT is 30 seconds.
+
+    local process timeout pidfile invalid
+
+    timeout=30
+    while [ "${1#-}" != "$1" ] && [ -n "$1" ]; do
+        case "$1" in
+            --timeout)
+                timeout="$2"
+                shift; shift
```

Comment:
annoying besserwisser nitpick: `shift 2` (we do require and only use bash) ?

Reply:
Good point, will change.